core/rawdb/accessors_metadata.go

```diff
@@ -79,3 +80,58 @@ func WriteChainConfig(db ethdb.KeyValueWriter, hash common.Hash, cfg *params.Cha
 		log.Crit("Failed to store chain config", "err", err)
 	}
 }
+
+// ucmList is a list of unclean-shutdown-markers, for rlp-encoding to the
+// database
+type ucmList struct {
+	Discarded uint64   // how many ucs have we deleted
+	Recent    []uint64 // unix timestamps of 10 latest unclean shutdowns
+}
+
+// UpdateUncleanShutdownMarker appends a new unclean shutdown marker and returns
+// the previous data
+// - a list of timestamps
+// - a count of how many old unclean-shutdowns have been discarded
+func UpdateUncleanShutdownMarker(db ethdb.KeyValueStore) ([]uint64, uint64, error) {
```

Comment:
Maybe `PushUncleanShutdownMarker` is better? It's basically a stack as the container.